aapp_runner/helper_functions.py

```diff
@@ -30,6 +30,63 @@
 LOGGER = logging.getLogger(__name__)
 
 
+def check_if_scene_is_unique(config):
+    """Check if the Scene is unique.
+
+    The scene is checked against the register (holding already processed
+    scenes).  If it overlaps in time with a previously processed scene and the
+    area-id's (used to collect the data (granules) are the same, then return
+    False - the scene is then not unique and should not be processed further.
+
+    """
+    LOGGER.debug("config.job_register: %s", str(config.job_register))
+    LOGGER.debug("config platform_name: %s", str(config['platform_name']))
+    LOGGER.debug("config - collection_area_id: %s", str(config['collection_area_id']))
+
+    # Use sat id, start and end time and area_id as the unique identifier of the scene!
+    if (config['platform_name'] in config.job_register and
+            len(config.job_register[config['platform_name']]) > 0):
+
+        # Go through list of start,end time tuples and see if the current
+        # scene overlaps with any - only if the area ids are the same
+
+        # Get registered start and end times with area id equal to current area_id
+        registered_times = []
+        for start_t, end_t, area_id in config.job_register[config['platform_name']]:
+            if area_id == config['collection_area_id']:
+                registered_times.append((start_t, end_t))
```

Comment:
How does this scale when the runner has a high uptime?
Maybe use a [deque](https://docs.python.org/3/library/collections.html#collections.deque) with a maximum length instead of a list?

Reply:
Is it placing the queue of recently processed IDs in the config object?  That seems a bit maluse and confusing.  Maybe this needs a dedicated type?

Reply:
I think this is cleaned after some time. So it will not build up.

For the config object it may be confusing,. But it is quite convenient to everything you need for controlling your processing in one object.

Reply:
Yes, there is a `reset_job_registry` and this can be configured via `locktime_before_rerun`
I don't use or know deque, but will consider in the future.